jena-arq/src/main/java/org/apache/jena/sparql/core/mem/DatasetGraphInMemory.java

```diff
@@ -55,21 +51,15 @@
 
     private final DatasetPrefixStorage prefixes = new DatasetPrefixStorageInMemory();
 
+    /** This lock imposes the multiple-reader and single-writer policy of transactions */
     private final Lock writeLock = new LockMRPlusSW();
 
-    private Lock writeLock() {
-        return writeLock;
-    }
-
-    private final ReentrantReadWriteLock commitLock = new ReentrantReadWriteLock(true);
-
     /**
-     * Commits must be atomic, and because a thread that is committing alters the various indexes one after another, we
-     * lock out {@link #begin(ReadWrite)} while {@link #commit()} is executing.
+     * Transaction lifecycle operations must be atomi, especially begin and commit where
```

Comment:
This comment is a bit confusing. How about:
```
We have a lock for committing because transaction lifecycle operations must be atomic, especially {@link Transactional#begin} and {@link Transactional#commit}, for which the global state of the (possibly multiple) indexes may be read or written one after another.
```